solidity/ecdsa/test/WalletRegistry.Upgrade.test.ts

```diff
@@ -0,0 +1,336 @@
+        })
+      })
+
+      describe("when a variable was removed", () => {
+        it("should throw an error", async () => {
+          await deployments.fixture()
+
+          await expect(
+            upgradeProxy("WalletRegistry", "WalletRegistryV2MissingSlot", {
+              factoryOpts: {
+                libraries: { EcdsaInactivity: EcdsaInactivity.address },
+                signer: proxyAdminOwner,
+              },
+              proxyOpts: {
+                constructorArgs: [AddressZero, AddressZero],
+                unsafeAllowLinkedLibraries: true,
+              },
+            })
+          ).to.be.rejectedWith(
+            Error,
+            "Deleted `_maliciousDkgResultNotificationRewardMultiplier`"
+          )
+        })
+      })
+    })
+
+    describe("when a new contract is valid", () => {
+      let tokenStaking: Contract
+      let reimbursementPool: Contract
+      let walletRegistryGovernance: Contract
+      let walletRegistry: WalletRegistry
+      let newWalletRegistry: WalletRegistryV2
+
+      const newSortitionPoolAddress =
+        "0x0000000000000000000000000000000000000101"
+      const newRandomBeaconAddress =
+        "0x0000000000000000000000000000000000000202"
+      const newVarValue = "new variable set for new contract"
+
+      before(async () => {
+        await deployments.fixture()
+
+        tokenStaking = await ethers.getContract("TokenStaking")
+        reimbursementPool = await ethers.getContract("ReimbursementPool")
+        walletRegistryGovernance = await ethers.getContract(
+          "WalletRegistryGovernance"
+        )
+        walletRegistry = (await ethers.getContract(
+          "WalletRegistry"
+        )) as WalletRegistry & WalletRegistryV2
+
+        expect(await walletRegistry.governance()).equal(
+          walletRegistryGovernance.address
+        )
+
+        newWalletRegistry = (await upgradeProxy(
+          "WalletRegistry",
+          "WalletRegistryV2",
+          {
+            factoryOpts: {
+              signer: proxyAdminOwner,
+              libraries: { EcdsaInactivity: EcdsaInactivity.address },
+            },
+            proxyOpts: {
+              constructorArgs: [newSortitionPoolAddress, tokenStaking.address],
+              call: {
+                fn: "initializeV2",
+                args: [newRandomBeaconAddress, newVarValue],
+              },
+              unsafeAllowLinkedLibraries: true,
+            },
+          }
+        )) as WalletRegistryV2
+      })
+
+      it("new instance should have the same address as the old one", async () => {
+        expect(newWalletRegistry.address).equal(walletRegistry.address)
+      })
+
+      it("should not update governance", async () => {
+        expect(await walletRegistry.governance()).equal(
+          walletRegistryGovernance.address
+        )
+      })
+
+      it("should reinitialize existing immutable variable", async () => {
```

Comment:
I think the name of this test is not correct. The value of an immutable variable is stored in the bytecode of the implementation contract. Nothing in the storage of the proxy or implementation contract is updated. Especially, nothing is reinitialized (`initializer` / `reinitializer` modifiers).

Maybe: "should use the new value of the immutable variable"?

Reply:
To add more context, the very next test name is correct because the random beacon address is passed as a parameter to `initializeV2` function which is `reinitializer(2)`.

Reply:
96cae7abe3b6d5b63e53817845ecb5a2a10c641e